src/code/z_parameter.c

```diff
@@ -2564,49 +2873,568 @@ s16 D_801BFC40[] = {
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80119610.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_8011B4E0.s")
+void Interface_SetPerfectLetters(PlayState* play, s16 perfectLettersType) {
+    InterfaceContext* interfaceCtx = &play->interfaceCtx;
+    s16 i;
 
-u16 D_801BFC50[] = {
-    0xC000, 0xE000, 0x0000, 0x2000, 0xA000, 0x8000, 0x6000, 0x4000,
+    interfaceCtx->perfectLettersOn = true;
+    interfaceCtx->perfectLettersType = perfectLettersType;
+
+    interfaceCtx->perfectLettersPrimColor[0] = 255;
+    interfaceCtx->perfectLettersPrimColor[1] = 165;
+    interfaceCtx->perfectLettersPrimColor[2] = 55;
+    interfaceCtx->perfectLettersPrimColor[3] = 255;
+    interfaceCtx->perfectLettersColorTimer = 20;
+
+    interfaceCtx->perfectLettersColorTargetIndex = 0;
+    interfaceCtx->perfectLettersCount = 1;
+    interfaceCtx->perfectLettersTimer = 0;
+
+    for (i = 0; i < PERFECT_LETTERS_NUM_LETTERS; i++) {
+        interfaceCtx->perfectLettersAngles[i] = 0;
+        interfaceCtx->perfectLettersState[i] = interfaceCtx->perfectLettersOffsetX[i] = 0;
+        if (interfaceCtx->perfectLettersType == PERFECT_LETTERS_TYPE_1) {
+            interfaceCtx->perfectLettersSemiAxisX[i] = 140.0f;
+            interfaceCtx->perfectLettersSemiAxisY[i] = 100.0f;
+        } else {
+            interfaceCtx->perfectLettersSemiAxisY[i] = 200.0f;
+            interfaceCtx->perfectLettersSemiAxisX[i] = 200.0f;
+        }
+    }
+    interfaceCtx->perfectLettersState[0] = PERFECT_LETTERS_STATE_INIT;
+}
+
+u16 sPerfectLettersType1AngleOffScreenTargets[PERFECT_LETTERS_NUM_LETTERS] = {
```

Comment:
Firstly I'm surprised that this of all things is u16. Secondly I feel like we probably could shorten this ... somehow.

Reply:
The `u16` is weird, I double checked and yeah it's loading in an unsigned half word in the function. 

I could see it being okay to drop "Targets" but then it does lose a little consistency

Reply:
I think it'll be okay with a comment attached, we have to draw the line somewhere.